src/typeEvaluator/functions.ts

```diff
@@ -296,6 +296,9 @@ export function handleFuncCallNode(node: FuncCallNode, scope: Scope): TypeNode {
         })
       })
     }
+    case 'sanity.versionOf': {
+      return {type: 'array', of: {type: 'string'}}
+    }
```

Comment:
Ideally we would also check the input to ensure it's a string:

```suggestion
    case 'sanity.versionOf': {
      const typeNode = walk({node: node.args[0], scope})
      return mapConcrete(typeNode, scope, (typeNode) => {
        if (typeNode.type !== 'string') {
          return {type: 'null'}
        }
      	return {type: 'array', of: {type: 'string'}}
      })
    }
```

this can help if you use `versionOf(someField)` and `someField` is not a string